eclair-core/src/test/scala/fr/acinq/eclair/payment/relay/ChannelRelayerSpec.scala

```diff
@@ -68,11 +68,12 @@ class ChannelRelayerSpec extends ScalaTestWithActorTestKit(ConfigFactory.load("a
     fwd
   }
 
-  def expectFwdAdd(register: TestProbe[Any], channelId: ByteVector32, outAmount: MilliSatoshi, outExpiry: CltvExpiry): Register.Forward[CMD_ADD_HTLC] = {
+  def expectFwdAdd(register: TestProbe[Any], channelId: ByteVector32, outAmount: MilliSatoshi, outExpiry: CltvExpiry, outConfidence: Double): Register.Forward[CMD_ADD_HTLC] = {
     val fwd = register.expectMessageType[Register.Forward[CMD_ADD_HTLC]]
     inside(fwd.message) { case add: CMD_ADD_HTLC =>
       assert(add.amount == outAmount)
       assert(add.cltvExpiry == outExpiry)
+      assert(add.confidence == outConfidence)
```

Comment:
It's a bit weird and incomplete to test confidence values here instead of endorsement values: it doesn't test the conversion from endorsement to confidence and from confidence to endorsement and the values don't make much sense to read.

What about:

- creating the incoming `update_add_htlc` from an endorsement value (`createValidIncomingPacket` takes an `endorsement: Byte` and directly creates an `Endorsement` tlv in the `UpdateAddHtlc`)
- change `expectFwdAdd` to take an `outEndorsement: Byte` parameter, which verifies that we correctly relay the incoming endorsement by creating the `UpdateAddHtlc` from the `CMD_ADD_HTLC` (`val htlc = UpdateAddHtlc(randomBytes32(), 0, add.amount, add.paymentHash, add.cltvExpiry, add.onion, add.nextBlindingKey_opt, add.confidence)`) and checks its `Endorsement` tlv

Reply:
Done